test/remote/gateways/remote_optimal_payment_test.rb

```diff
@@ -57,6 +58,52 @@ def test_successful_verify
     assert_equal 'no_error', response.message
   end
 
+  def test_stored_data_auth_and_capture_after_store
+    response = @gateway.store(@credit_card, @options)
+    assert_success response
+    assert_equal 'no_error', response.message
+
+    assert auth = @gateway.stored_authorize(@amount, response.authorization)
+    assert_success auth
+    assert_equal 'no_error', auth.message
+    assert auth.authorization
+
+    assert capture = @gateway.capture(@amount, auth.authorization)
+    assert_success capture
+  end
+
+  def test_stored_data_purchase_after_store
+    response = @gateway.store(@credit_card, @options)
+    assert_success response
+    assert_equal 'no_error', response.message
+
+    assert stored_purchase = @gateway.stored_purchase(@amount, response.authorization)
+    assert_success stored_purchase
+  end
+
+  def test_stored_data_auth_after_failed_store
+    response = @gateway.store(@invalid_card, @options)
+    assert_failure response
+    assert_not_nil response.authorization
+    assert_equal 'ERROR', response.params['decision']
+
+    assert auth = @gateway.stored_authorize(@amount, response.authorization)
+    assert_failure auth
+    assert_equal 'ERROR', auth.params['decision']
+  end
+
+  def test_stored_data_purchase_after_failed_store
+    response = @gateway.store(@invalid_card, @options)
+    assert_failure response
+    assert_not_nil response.authorization
+    assert_equal 'ERROR', response.params['decision']
+
+
+    assert stored_purchase = @gateway.stored_purchase(@amount, response.authorization)
+    assert_failure stored_purchase
+    assert_equal 'ERROR', stored_purchase.params['decision']
+  end
```

Comment:
@curiousepic how do these last two seem for failed test cases? I don't see a specific invalid test card, so just created a card with a blank PAN for these two

Reply:
👍